alvr/server_openvr/src/lib.rs

```diff
@@ -383,24 +383,30 @@ extern "C" fn report_present(timestamp_ns: u64, offset_ns: u64) {
 }
 
 extern "C" fn wait_for_vsync() {
+    // Default 120Hz-ish wait if StatisticsManager isn't up.
+    // We use 120Hz-ish so that SteamVR doesn't accidentally get
+    // any weird ideas about our display Hz with its frame pacing.
+    static PRE_HEADSET_STATS_WAIT_INTERVAL: Duration = Duration::from_millis(8);
+
     // NB: don't sleep while locking SERVER_DATA_MANAGER or SERVER_CORE_CONTEXT
-    let sleep_duration = if alvr_server_core::settings()
-        .video
-        .enforce_server_frame_pacing
-    {
-        SERVER_CORE_CONTEXT
-            .read()
-            .as_ref()
-            .and_then(|ctx| ctx.duration_until_next_vsync())
-    } else {
-        None
-    };
+    let sleep_duration = SERVER_CORE_CONTEXT
+        .read()
+        .as_ref()
+        .and_then(|ctx| ctx.duration_until_next_vsync());
 
     if let Some(duration) = sleep_duration {
-        thread::sleep(duration);
+        if alvr_server_core::settings()
+            .video
+            .enforce_server_frame_pacing
+        {
+            thread::sleep(duration);
+        } else {
+            thread::yield_now();
```

Comment:
Btw what's the thought behind yielding vs just not doing anything?

Reply:
SteamVR is a fairly high priority process so a straight busy loop is probably not a great idea, I guess. Also scheduling consistency between the two modes, so (hopefully) other Rust threads wouldn't get starved if they exist.